js/src/utils.js

```diff
@@ -248,6 +271,37 @@ export function lastBytes(arr, num) {
   return safeSlice(arr, arr.length - num);
 }
 
+export function getErrMsg(e) {
+  switch (e.message) {
+    case 1:
+      return [false, errors.ERR_INVALID_CHAIN]
```

Comment:
What is the purpose of returning the array with false as the first argument?

Reply:
Different consensus processes may handle `throw`n errors different way. By using a verry explicit non-`throw` error pattern, we can give developers a bit more leeway. Basically a library policy of "we will not throw" like in golang.

Reply:
I'm not sure avoiding `throw` is what we want to do. If we `throw new Error()` then the user is able to handle it using `try/catch` blocks.

```js
function useValidateHeader (headers) {
  try {
    ValidateSPV.validateHeader(headers)
  } catch (e) {
    // handle error
  }
}
```   

If we `return new Error()`, then the user has to check if the value is an instance of an error.

```js
function useValidateHeader (headers) {
  if (ValidateSPV.validateHeader(headers) instanceof Error) {
    // handle error
  }
}
```

Or, not return/throw an Error object and use the `[valid, reason]` pattern. If this, then all the return values should follow this pattern e.g. `[Boolean, Any]` 

```js
function useValidateHeader (headers) {
  const [valid, reason] = ValidateSPV.validateHeader(headers)
  if (valid) {
    // handle valid
  } else {
    // handle error
}
```